File: baekjoon/1874/pon04178/1874.c
#include <stdio.h>
#include <stdlib.h>

int main() {
	int n, result_n = 0;
	int top = -1, t = 1;
	scanf("%d", &n);

	int* input = (int*)malloc(sizeof(int) * n);
	int* stack = (int*)malloc(sizeof(int) * n);
	char* result = (char*)malloc(sizeof(char) * n * 2);

	for (int i = 0; i < n; i++)
		scanf("%d", &input[i]);

	for (int i = 0; i < n; i++) {
		while (result_n < n * 2) {
			if (stack[top] == input[i]) {
				stack[top--] = 0;
				result[result_n++] = '-';
				break;
			}
			stack[++top] = t++;
			result[result_n++] = '+';
		}
	}

	if (top != -1) {
		printf("NO");
	}
	else {
		for (int i = 0; i < n * 2; i++)
			printf("%c\n", result[i]);
	}
	return 0;
}

File: baekjoon/data_structure/1874/1874.py
import sys
from collections import deque

sys.stdin = open("baekjoon\data_structure\\1874\input.txt")
input = sys.stdin.readline

n = int(input())
stack = deque()
result = deque()
top = 1

for i in range(n):
    num = int(input())
    while top <= num:
        stack.append(top)
        result.append('+')
        top += 1
    
    if stack[-1] == num:
        stack.pop()
        result.append('-')
        
    else:
        print("NO")
        break

else:
    for i in result:
        print(i)